### PR TITLE
Defect 58/oauth2 server checks account verified flag before password on an account

### PR DESCRIPTION
**Pull request by andrewmil**
Now checking if the user is verified/locked after checking the user's password

Added a couple of extra unit tests for this part of verifying the user

### CODE REVIEW

**Comment by tej99** [comment]
Is it possible for a user to be blocked before the account if verified? If so maybe have the account_locked check before account_is_verified. Besides that LGTM.

**Comment by andrewmil** [comment]
The order we have these in determines if that is possible I think, if we have the verified check first it wont happen

EDIT: Actually I'm not sure about this one, we probably want business clarification on this

File: oauth2server/apps/tokens/tests/test_user_credentials.py [hunk 1]
```diff
@@ -36,16 +32,10 @@ def test_missing_client_credentials(self):
 
         self.assertEqual(OAuthAccessToken.objects.count(), 0)
         self.assertEqual(OAuthRefreshToken.objects.count(), 0)
-
         self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
         self.assertEqual(response.data['detail'], u'Client credentials were not found in the headers or body')
```
Comment (nherriot) [inline oauth2server/apps/tokens/tests/test_user_credentials.py:36]: There is a print statement that could be removed.

**Review by nherriot — CHANGES_REQUESTED** [review]
Some print statements in test_client_credentials.py that could be removed. They are not part of what you changed but I happened to stumble across them. Readme needs to be updated with Python version.

**Pull request merged**